### DISCUSSION TITLE
Should Audacious use GTK 2 or GTK 3 by default?

### DISCUSSION THREAD

**Discussion by radioactiveman** [audacious-media-player/audacious#134] [Polls]
With Audacious 4.3 we have restored the support of GTK 3 but use GTK 2 by default.

Distributions like Debian, Ubuntu and OpenSUSE enable GTK 3 support meanwhile.
Fedora sticks to our defaults and uses GTK 2.
Arch Linux and Gentoo disable GTK support, probably because GTK 2 has reached its end-of-life.

I would like to get some feedback which GTK version Audacious should use by default and why.
We then may change this with the upcoming Audacious version 4.4.

Thank you.

**Comment by y0shir** [comment]
I think it should stick with GTK2 on windows and move to GTK3 on linux.

**Reply by birdie-github** [reply]
GTK2 is just dead, GTK3 is still kinda sort-of supported.

I've voted for "Switch to GTK3".

**Reply by jlindgren90** [reply]
@y0shir just curious, do you use the GTK (2) interface on Windows? I don't have a problem with continuing to support it there if people are using it, but we have certainly put more effort lately into the Qt interface, and to me it provides a better experience.

**Comment by ffs-akl** [comment]
unbeliveable！ it's 2024, and you're still discussing whether to use GTK2 as the default!

**Reply by y0shir** [reply]
Why not? Does GTK2 magically stop working past certain date?

**Reply by ffs-akl** [reply]
Of course not, so that you can compile your own special GTK2 version with -enable gtk2

**Comment by radioactiveman** [comment]
My arguments for making GTK 3 the default are:

- GTK 2 is end-of-life and phased out eventually by distributions
- Several distributions (even big ones like Debian) therefore provide Audacious as GTK 3 build
- Audacious 4.4 would then (by default) make use of maintained toolkits again with the switch to Qt 6 / GTK 3
- Better look in modern environments with symbolic icons and same color for menu bar and toolbar
- Better support for High DPI screens
- Support for Wayland
- The issues/pain we had years ago with GTK 3 (like themes breaking with every release, introduction of header bar, deprecating stuff all the time, ...) is no longer the case since the upstream focus switched to GTK 4
- All votes here are in favor of GTK 3 (IIRC @y0shir voted for GTK 2, but stated as comment that Linux should use GTK 3)

@jlindgren90: Could you please reconsider your decision? I would be glad if we can make this switch before releasing 4.4-beta.
For the Windows build it would be fine for me to stay at GTK 2 or make it even Qt-only. Thanks :crossed_fingers:

**Reply by jlindgren90** [reply]
The main thing to me is that you are now doing most of the work maintaining Audacious (and especially on the GTK side). So it's right that you should make the decision. Following what distributions are doing by default is also reasonable.

Personally, I still disagree with many GTK 3 design decisions, which to me are anti-user. The push for monochrome icons (or outright disabling icons in favor of text-only menus/buttons) hurts usability for the sake of some designer's preferred "aesthetic". Choices like that should be left to the user. But we have (and I prefer) the Qt interface, which is more user-friendly in this regard.

I want to temper expectations regarding (native) Wayland support. The current set of Wayland protocols intentionally break existing applications by forbidding application-side window positioning. Audacious (and especially the Winamp interface) needs this to work properly. There is a proposal for a new Wayland protocol to fix this, but until that's approved and widely supported, I think we should recommend that people run Audacious via XWayland, regardless of toolkit.

**Reply by radioactiveman** [reply]
Alright, I changed now the default GTK version for Audacious 4.4 to version 3. Thanks for the appreciation.

Our summary at https://audacious-media-player.org/problems now includes this recommendation regarding Wayland.

**Comment by rado84-github** [comment]
Is there any way to switch to gtk3 mode in audacious 4.1 without recompiling? Unfortunately I can't update it bc OS is Mint and I'm gonn fall into a dependency hell, if I tried that.

**Reply by ffs-akl** [reply]
if your audacious package  was compilied with ` -enable gtk3`,  you can switch to gtk3 mode in the menu of audacious.  
otherwise you will have to recompile it  with ` -enable gtk3`  for gtk3 mode.
